File: ambeth/Ambeth.Cache/ambeth/cache/rootcachevalue/DefaultRootCacheValue.cs
using De.Osthus.Ambeth.Merge.Model;
using System;

namespace De.Osthus.Ambeth.Cache.Rootcachevalue
{
	public class DefaultRootCacheValue : RootCacheValue
	{
        protected readonly IEntityMetaData metaData;

		protected Object[] primitives;

		protected IObjRef[][] relations;

		private Object id;

		private Object version;

		public DefaultRootCacheValue(IEntityMetaData metaData) : base(metaData)
		{
            this.metaData = metaData;
		}

        public override Object Id
        {
            get
            {
                return id;
            }
            set
            {
                this.id = value;
            }
        }

        public override Object Version
        {
            get
            {
                return version;
            }
            set
            {
                this.version = value;
            }
        }
		
		public override void SetPrimitives(Object[] primitives)
		{
			this.primitives = primitives;
		}

		public override Object[] GetPrimitives()
		{
			return primitives;
		}

        public override object GetPrimitive(int primitiveIndex)
        {
            return primitives[primitiveIndex];
        }

		public override IObjRef[][] GetRelations()
		{
			return relations;
		}

		public override void SetRelations(IObjRef[][] relations)
		{
			this.relations = relations;
		}

        public override IObjRef[] GetRelation(int relationindex)
        {
            return relations[relationindex];
        }

		public override void SetRelation(int relationIndex, IObjRef[] relationsOfMember)
		{
			relations[relationIndex] = relationsOfMember;
		}

		public override Type EntityType
		{
            get
            {
                return metaData.EntityType;
            }
		}

        public override IEntityMetaData Get__EntityMetaData()
        {
            return metaData;
        }
	}
}

File: jambeth/jambeth-cache/src/main/java/de/osthus/ambeth/cache/rootcachevalue/DefaultRootCacheValue.java
package de.osthus.ambeth.cache.rootcachevalue;

import de.osthus.ambeth.merge.model.IEntityMetaData;
import de.osthus.ambeth.merge.model.IObjRef;

public class DefaultRootCacheValue extends RootCacheValue
{
	protected Object[] primitives;

	protected IObjRef[][] relations;

	protected Object id;

	protected Object version;

	protected IEntityMetaData metaData;

	public DefaultRootCacheValue(IEntityMetaData metaData)
	{
		super(metaData);
		this.metaData = metaData;
	}

	@Override
	public IEntityMetaData get__EntityMetaData()
	{
		return metaData;
	}

	@Override
	public Object getId()
	{
		return id;
	}

	@Override
	public void setId(Object id)
	{
		this.id = id;
	}

	@Override
	public Object getVersion()
	{
		return version;
	}

	@Override
	public void setVersion(Object version)
	{
		this.version = version;
	}

	@Override
	public void setPrimitives(Object[] primitives)
	{
		this.primitives = primitives;
	}

	@Override
	public Object[] getPrimitives()
	{
		return primitives;
	}

	@Override
	public Object getPrimitive(int primitiveIndex)
	{
		return primitives[primitiveIndex];
	}

	@Override
	public IObjRef[][] getRelations()
	{
		return relations;
	}

	@Override
	public void setRelations(IObjRef[][] relations)
	{
		this.relations = relations;
	}

	@Override
	public IObjRef[] getRelation(int relationIndex)
	{
		return relations[relationIndex];
	}

	@Override
	public void setRelation(int relationIndex, IObjRef[] relationsOfMember)
	{
		relations[relationIndex] = relationsOfMember;
	}

	@Override
	public Class<?> getEntityType()
	{
		return metaData.getEntityType();
	}
}
